src/Microsoft.AspNetCore.Razor.Evolution/Legacy/HtmlMarkupParser.cs

```diff
@@ -789,8 +794,8 @@ private void BeforeAttribute()
             IEnumerable<HtmlSymbol> whitespaceAfterAttributeName)
         {
             // First, determine if this is a 'data-' attribute (since those can't use conditional attributes)
-            var name = nameSymbols.GetContent(Span.Start);
-            var attributeCanBeConditional = !name.Value.StartsWith("data-", StringComparison.OrdinalIgnoreCase);
+            var name = string.Concat(nameSymbols.Select(s => s.Content));
```

Comment:
I got rid of `GetContent`. All of these are good optimization opportunities for us.

For instance I don't think it's possible that the attribute name is multiple tokens here, but I didn't want to do too much.